### DISCUSSION TITLE
Keyboard shortcut to switch audio output to headphones

### DISCUSSION THREAD

**Discussion by AquiGorka** [johnlindquist/kit#1387] [Share]
Existing alias: `alias shead="SwitchAudioSource -s \"External Headphones\""` (uses [switchaudio-osx](https://github.com/deweller/switchaudio-osx))

Script:
```
// Name: Shortcut to Headphones
// Shortcut: cmd shift h
import "@johnlindquist/kit"
await $`/bin/zsh -lic shead`
```

**Comment by ndrake** [comment]
These are cool, great idea with the aliases.  I have a similar script that presents a list of all outputs (except the current one) to pick from.  I only really switch between two, so I may switch to a setup like yours with aliases and shortcuts. 👍 

```
/*
## Switch audio output device
*/

// Name: Switch Audio
// Description: Switch audio output device (Mac only)
// Author: Nate Drake

// Install SwitchAudioSource with `brew install switchaudio-osx`

import "@johnlindquist/kit"

if (!isMac) {
    await div(md('## Sorry.  I only work on Mac at the moment.'))
    exit()
}

const SwitchAudioSourcePath = '/opt/homebrew/bin/SwitchAudioSource'

const sasExists = await pathExists(SwitchAudioSourcePath)

if (sasExists) {

    const currentOutput = await $`${SwitchAudioSourcePath} -c`
    const items = await $`${SwitchAudioSourcePath} -a -t output`

    const choices = items.stdout.trim().split(/\r?\n/).filter(o => o !== currentOutput.stdout.trim())

    let output = await arg(
        {
            placeholder: 'Pick ouput device'
        },
        choices
    )

    await $`${SwitchAudioSourcePath} -s ${output} -t output`
} else {
    await div(md(`ERROR: Please install SwitchAudioSource`))
}
```

**Reply by AquiGorka** [reply]
Nice! I like your idea of having a shortcut to pick from a list, that could be helpful instead of having to hardcode every option as a shortcut

**Comment by AquiGorka** [comment]
I've added a notification when the audio source is changed: `alias shead="SwitchAudioSource -s \"External Headphones\" && osascript -e 'display notification \"Switched to headphones\" with title \"Audio source\"'"`
